js/glsl/scatter-vertex.glsl

```diff
@@ -50,50 +50,55 @@ attribute vec3 color_previous;
 void main(void) {
     vec3 origin = vec3(xlim.x, ylim.x, zlim.x);
     vec3 size_viewport = vec3(xlim.y, ylim.y, zlim.y) - origin;
+    vec3 animation_time = vec3(animation_time_x, animation_time_y, animation_time_z);
 
-#ifndef AS_LINE
+#ifdef AS_LINE
+    vec3 model_pos = (mix(position_previous, position, vec3(animation_time_x, animation_time_y, animation_time_z))
+                - origin) / size_viewport - 0.5;
+    vec4 view_pos = modelViewMatrix * pos;
+#else
     vec3 vector = v;
     vec3 vector_previous = v_previous;
     vec3 position_offset = vec3(x, y, z);
     vec3 position_offset_previous = vec3(x_previous, y_previous, z_previous);
 
     // assume the vector points to the y axis
-    vec3 vector_current = mix(normalize(vector_previous), normalize(vector), vec3(animation_time_vx, animation_time_vy, animation_time_vz))
+    vec3 vector_current = mix(normalize(vector_previous), normalize(vector), animation_time)
            * mix(length(vector_previous), length(vector), (animation_time_vx+ animation_time_vy+ animation_time_vz)/3.);
     vec3 y_axis = normalize(vector_current);
     // we may have bad luck, and alight with 1 vector, so take two vectors, and we'll always find a non-zero vector
     vec3 some_z_vector_a = vec3(0., 1., 1.);
     vec3 some_z_vector_b = normalize(vec3(0., 2., 1.));
     vec3 x_axis = normalize(cross(y_axis, some_z_vector_a)  + cross(y_axis, some_z_vector_b));
     vec3 z_axis = -normalize(cross(y_axis, x_axis)); // - to keep it right handed
-    float vector_length = length(vector_current);
+    //float vector_length = length(vector_current);
     // the following matrix should point it to the direction of 'vector'
     mat3 move_to_vector = mat3(x_axis, y_axis, z_axis);
-    //vec3 x = vec3(1, 0, 0);
-    //vec3 y = vec3(0, 1, 0);
-    //vec3 z = vec3(0, 0, 1);
-    //mat3 move_to_vector = mat3(z, y, x);
+
     float s = mix(size_previous/100., size/100., animation_time_size);
-    vec3 pos = (move_to_vector * (position*s))
-        + (mix(position_offset_previous, position_offset, vec3(animation_time_x, animation_time_y, animation_time_z))
-                - origin) / size_viewport - 0.5;
+    vec3 model_pos = (mix(position_offset_previous, position_offset, animation_time) - origin) / size_viewport - 0.5;
     //vec3 pos = (pos_object ) / size;// - 0.5;
-#else
-    vec3 pos = (mix(position_previous, position, vec3(animation_time_x, animation_time_y, animation_time_z))
-                - origin) / size_viewport - 0.5;
+    #ifdef USE_SPRITE
+        // if we are a sprite, we add the position in view coordinates, and need to 
+        vec4 view_pos = modelViewMatrix * vec4(model_pos, 1.0);
```

Comment:
USE_SPRITE doesn't need move_to_vector -- you can save some computation in that case (perhaps the compiler takes care of this already...)

Reply:
That was a nice refactor on the vertex shader.

Reply:
Actually, we could use this to rotate the sprites, what do you think? The z component could be used for maybe a stretch factor?

Reply:
That sounds like a great idea -- particularly for non-square textures...